src/flash-source-buffer.js

```diff
@@ -136,6 +174,10 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
       removeCuesFromTrack(0, Infinity, this.metadataTrack_);
       removeCuesFromTrack(0, Infinity, this.inbandTextTrack_);
     });
+
+    this.mediaSource_.player_.tech_.hls.on('dispose', () => {
```

Comment:
That would mean adding a `dispose` method to the flash and virtual buffer, which is not a standard api on mediasource buffers